### PR TITLE
Fix the status reporting of device to avoid issues defined by twin.

### PR DESCRIPTION
**Pull request by luomengY**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
In version v1.15.0, we removed the content of the user-defined status module. When deploying a device, the Twins reported by the device status are empty. In this case, we need to create Twins, otherwise the status will not have any content.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
@wbc6080 
**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wbc6080** [comment]
LGTM  thanks

**Review by WillardHu — CHANGES_REQUESTED** [review]

File: cloud/pkg/devicecontroller/controller/upstream.go [hunk 1]
```diff
@@ -226,10 +226,27 @@ func NewUpstreamController(dc *DownstreamController) (*UpstreamController, error
...
+				return twin
+			}
+			twin = &v1beta1.Twin{
+				PropertyName: twinName,
+			}
+			deviceStatus.Status.Twins = append(deviceStatus.Status.Twins, *twin)
+			return twin
+		}
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (luomengY) [reply cloud/pkg/devicecontroller/controller/upstream.go:243]: > The return value nil is not processed at the call when the twin doesn't match any properties, is that expected?

When twin returns to nil, a new twin will be created：
			......
			twin := findTwinByName(twinName, deviceStatus)
			if twin != nil {
				return twin
			}
			twin = &v1beta1.Twin{
				PropertyName: twinName,
			}
			......

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 2]
```diff
@@ -417,3 +420,20 @@ func NewDownstreamController(crdInformerFactory crdinformers.SharedInformerFacto
 	}
 	return dc, nil
 }
+
+// Remove twin with changed attribute names.
+func removeTwinWithNameChanged(device *v1beta1.Device) {
+	properties := device.Spec.Properties
+	twins := device.Status.Twins
+	var newTwins []v1beta1.Twin
```
[the remark this answers is not in the document]
Comment (luomengY) [reply cloud/pkg/devicecontroller/controller/downstream.go:428]: ok

File: cloud/pkg/devicecontroller/controller/upstream.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (WillardHu) [reply cloud/pkg/devicecontroller/controller/upstream.go:243]: ok

**Comment by wbc6080** [comment]
@Shelley-BaoYue @zhiyingfang2022 Please help add this pr to version 1.17, thanks

**Comment by zhiyingfang2022** [comment]
> @Shelley-BaoYue @zhiyingfang2022 Please help add this pr to version 1.17, thanks

ok,done.

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**